### ISSUE TITLE
Survey task filter radio button labels use nested buttons

### ISSUE THREAD

**Issue by eatyourgreens** [zooniverse/front-end-monorepo#3919]
## Package
lib-classifier

## Describe the bug
The label for each filter radio button includes a close button when the radio button is checked. `<label>` tags can only contain inline text and images. Nested `<button>` tags are invalid and raise an accessibility issue. With one clickable element nested inside another clickable element, the label's behaviour is not obvious when clicked.
https://github.com/zooniverse/front-end-monorepo/blob/699ef772c3a25a0bed9b51072898f0b4ad557eed/packages/lib-classifier/src/plugins/tasks/survey/components/components/Chooser/components/CharacteristicsFilter/components/FilterButton.js#L56-L65

The radio button's label is also confusing, since it contains the close button label.
<img width="707" alt="Radio button label saying: antelope/deer Remove antelope/deer filter." src="https://user-images.githubusercontent.com/59547/204101081-a5dc5305-80e5-42e1-a0f0-eb57e5ce0ce8.png">

## To Reproduce
Select a filter then inspect the radio button label in the DOM.

## Expected behavior
Labels should use valid HTML and not generate accessibility errors.

**Comment by mcbouslog** [comment]
I think the close / remove button should be removed.

With the desired UI of selecting or deselecting one item, and only ever one item at a time, from a group of items, what's the best implementation?

Maybe a checkbox group with that only allows one item to be checked?
Maybe a radio button group with onClick that unchecks checked inputs?
Something else?

**Comment by eatyourgreens** [comment]
>Maybe a checkbox group with that only allows one item to be checked?

This is basically re-inventing radio buttons, albeit with the option to turn one off. Checkboxes have a completely different keyboard interaction from radio buttons, so the design would have to clarify, visually, that the interaction is different.

**Comment by eatyourgreens** [comment]
We could also allow multiple selection. See, for example, the filtered search on https://www.johnlewis.com/search?search-term=coffee

**Comment by eatyourgreens** [comment]
Radio buttons are a good choice when only one answer can be selected but always provide an extra option eg. Male/Female/Prefer not to answer.

**Comment by eatyourgreens** [comment]
The [GDS guidelines for radio buttons](https://design-system.service.gov.uk/components/radios/) are worth reading.

**Comment by mcbouslog** [comment]
Not suggesting we mirror, but for reference here's [what PFE does](https://github.com/zooniverse/Panoptes-Front-End/blob/master/app/classifier/tasks/survey/chooser/CharacteristicsFilter.jsx#L71-L77). PFE provides a Clear button that's only enabled once a different option within the related characteristic group has been selected, and when Clear is clicked it clears and closes the related filter section.

Project with survey task - https://master.pfe-preview.zooniverse.org/projects/markb-panoptes/survey-task-test-project/classify?reload=0&workflow=2434&env=staging

The linked GDS guidelines note:
"Users cannot go back to having no option selected once they have selected one, without refreshing their browser window. Therefore, you should include ‘None of the above’ or ‘I do not know’ if they are valid options."

I'm not sure, but I think the two options to consider are:
a) changing the characteristic section to a checkbox group and allowing the selection of multiple specific values (i.e. red AND black)
b) keeping the characteristic section a radio group and adding a "no filter" option

@seanmiller26 - what are your thoughts?

**Comment by seanmiller26** [comment]
I have reconfigured the functionality to be checkboxes instead of radio buttons. 
<img width="470" alt="Screen Shot 2022-12-08 at 1 31 48 PM" src="https://user-images.githubusercontent.com/89043489/206550477-99e63de2-c010-4e28-9014-3133e70909e0.png">
This allows volunteers make multiple selections. Notice a tally counts the active filters next to the 'Filters' button and a 'Clear All Filters' button only shows when a selection is made. There is also an overlay applied to the checkboxes to make the checks more visible.
<img width="466" alt="Screen Shot 2022-12-08 at 1 33 28 PM" src="https://user-images.githubusercontent.com/89043489/206550479-d42e1cc0-bc80-40bf-b7dd-dc9814bd5e0f.png">
Here is how it looks after the selections are made and the filter dialog closes. Volunteers can see there are filters applied but would have to go back into the filter dialog and choose new selections, similar to PFE's current functionality. 
<img width="466" alt="Screen Shot 2022-12-08 at 1 33 37 PM" src="https://user-images.githubusercontent.com/89043489/206550480-cc79bb80-c4a0-46d7-9b57-e10e3f6a3591.png">
Let me know if there are other functionalities I have missed.
